### PR TITLE
Cache updater fixes

### PR DESCRIPTION
**Pull request by skovhus**
Follow up on the changes in https://github.com/tmrowco/electricitymap-contrib/pull/3216.

This PR fixes the emission graphs and the history mix graph. 

I recommend reviewing with the w=1 query parameter set: https://github.com/tmrowco/electricitymap-contrib/pull/3301/files?w=1

Ref: https://github.com/tmrowco/electricitymap/issues/1225

### CODE REVIEW

File: web/src/reducers/dataReducer.js [hunk 1]
```diff
@@ -182,8 +181,6 @@ module.exports = (state = initialDataState, action) => {
           [action.zoneId]: action.payload.map(datapoint => ({
             ...datapoint,
             hasParser: true,
-            // Exchange information is not shown in history observations without production data, as the percentages are incorrect
-            exchange: isEmpty(datapoint.production) ? {} : datapoint.exchange,
```
Comment (madsnedergaard) [inline web/src/reducers/dataReducer.js:186]: What's the reason for removing `exchange` here altogether? Where did it show? :)

Comment (madsnedergaard) [reply web/src/reducers/dataReducer.js:186]: Ah nevermind, I now realise we already do `...datapoint` above!

**Review by FelixDQ — APPROVED** [review]
🙌

**Review by madsnedergaard — APPROVED** [review]

File: web/src/reducers/dataReducer.js (hunk 1 above)
Comment (skovhus) [reply web/src/reducers/dataReducer.js:186]: `exchange` is still there, but now we do not modify it in the reducer. Less remapping.

**Pull request merged**